portal-api/src/external-api/currencies.js

```diff
@@ -24,16 +26,33 @@ const getCurrencies = async () => {
 };
 
 const getCurrency = async (id) => {
+  if (!isValidRegex(CODE, id)) {
```

Comment:
Validator.js has an `isISO4217()` function to check currencies that we can use here and will be a bit stricter